spec/controllers/relationships_controller_spec.rb

```diff
@@ -0,0 +1,135 @@
+require "spec_helper"
+
+describe RelationshipsController do
+  let(:user) { Fabricate(:user) }
+  let(:john) { Fabricate(:user) }
+
+  describe 'GET #index' do
+    it_behaves_like "requires sign in" do
+      let(:action) { get :index }
+    end
+
+    context 'authenticated users' do
+      before { sign_in(user) }
+
+      it "should set empty @relationships variable when there are no relationships" do
+        get :index
+
+        expect(assigns(:relationships)).to eq([])
+      end
+
+      it 'should set @relationships variable to the following relationships' do
+        relationship = Fabricate(:relationship, leader: john, follower: user)
+
+        get :index
+
+        expect(assigns(:relationships)).to eq [ relationship ]
+      end
+
+      it 'should render the index template' do
+        get :index
+
+        expect(response).to render_template 'index'
+      end
+
+    end
+  end
+
+  describe 'POST #create' do
+    it_behaves_like "requires sign in" do
+      let(:action) { post :create, leader_id: 1 }
+    end
+
+    context 'authenticated users' do
+      before { sign_in(user) }
+
+      it 'creates the Relationship' do
+        post :create, leader_id: john.id
+
+        expect(Relationship.count).to eq 1
+      end
+
+      it 'creates a relationship that the current user follows the leader' do
+        post :create, leader_id: john.id
+
+        expect(user.following_relationships.first.leader).to eq john
+      end
+
+      it 'does not create a relationship if the current user is already following the leader' do
+        Fabricate(:relationship, leader: john, follower: user)
+
+        post :create, leader_id: john.id
+
+        expect(Relationship.count).to eq 1
+      end
+
+      it 'does not create a relationship if the current user is trying to follow themselves' do
+        post :create, leader_id: user.id
+
+        expect(Relationship.count).to eq 0
+      end
+
+      it 'redirects to the people page' do
+        post :create, leader_id: john.id
+
+        expect(response).to redirect_to people_path
+      end
+
+      it 'shows a flash message' do
+        post :create, leader_id: john.id
+
+        expect(flash[:success]).to be_present
+      end
+
+      it 'throws an ActiveRecord::RecordNotFound exception when the user cant be found' do
```

Comment:
I would probably catch this error and gracefully return a failure message to the user, rather than have them see the 500 page.